### PR TITLE
Rework displaying HDF5 datatypes

### PR DESCRIPTION
**Pull request by takluyver**
Previously, h5glance relied on converting HDF5 datatypes to numpy dtypes (in h5py) before trying to display them. This works OK for normal datatypes (e.g. int32, float64), but HDF5 allows for much more flexibility, like non-standard float formats, or using 3 bytes per number. Converting to numpy dtypes would either break in these cases, or 'promote' it to a standard dtype and give misleading information about what's in the file. I don't think these are used often, but I want h5glance to do something sensible if they come up.

This works directly from the h5py TypeID objects instead. Standard types are still displayed as e.g. `uint64`, while non-standard numeric types will look like `custom 3-byte float`. I've gone through the TypeID subclasses and tried to cover all the possible datatypes.

### CODE REVIEW

**Comment by takluyver** [comment]
Heads up @vallsv - this probably breaks things for non-h5py objects. I'm open to addressing that, but the usual constraints apply: I have limited time, and my focus is on using this with h5py objects.

As I learned about HDF5, I realised that its datatypes don't necessarily map directly to numpy dtypes, so the focus of this is to describe HDF5 datatypes rather than numpy dtypes.

File: h5glance/datatypes.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        un = 'un' if hdf_dt.get_sign() == h5t.SGN_NONE else ''
+        return "{}-byte {}signed integer".format(size, un)
+    elif isinstance(hdf_dt, h5t.TypeFloatID):
+        # Check normal float dtypes first
+        for candidate, descr in float_types_by_size().get(size, ()):
+            if hdf_dt == candidate:
+                return descr
+        return "custom {}-byte float".format(size)
+    elif isinstance(hdf_dt, h5t.TypeBitfieldID):
+        return "{}-byte bitfield"
```
Comment (tmichela) [inline h5glance/datatypes.py:51]: you forgot to format the number of bytes here (same for `TypeOpaque`)

Comment (tmichela) [inline h5glance/datatypes.py:51]: you forgot to format the number of bytes here (same for TypeOpaque)

**Comment by tmichela** [comment]
I'm not expert in h5py types, but that looks good to me

File: h5glance/datatypes.py (hunk 1 above)
Comment (takluyver) [reply h5glance/datatypes.py:51]: Thanks, well caught!

**Pull request merged**